apps/condo/domains/organization/components/EmployeeRolesSettingsContent.tsx

```diff
@@ -1,62 +1,629 @@
+            let otherEmployeeHasPermission = false
+
+            for (const roleId of otherEmployeeRoleIds) {
+                const isEmployeeHasPermission = get(permissionsState, [roleId, ...pathToPermissionsGroupFromEmployeeRole, permission], false)
+
+                if (isEmployeeHasPermission) {
+                    otherEmployeeHasPermission = true
+                    break
+                }
+            }
+
+            if (!otherEmployeeHasPermission) {
+                checkboxDisabled = true
+                break
+            }
+        }
+    }
+
+    return checkboxDisabled
+}
+
+/**
+ * Returns a message notifying that the role update is complete.
+ * If there is only a checkbox withdrawal (removal of permissions), then there will be a DeletePermissionsMessage message,
+ * in another case (adding permissions/adding and deleting permissions) will be a UpdatePermissionsMessage message.
+ */
+const getSaveNotificationMessage = ({ intl, initialPermissionsState, permissionsState }): string => {
+    const UpdatePermissionsMessage = intl.formatMessage({ id: 'pages.condo.settings.employeeRoles.notification.updatePermissions' })
+    const DeletePermissionsMessage = intl.formatMessage({ id: 'pages.condo.settings.employeeRoles.notification.deletePermissions' })
+
+    let notificationMessage = DeletePermissionsMessage
+
+    for (const roleId of Object.keys(initialPermissionsState)) {
+        const initialAppsPermissions = initialPermissionsState[roleId].b2bAppRoles
+        const newAppsPermissions = permissionsState[roleId].b2bAppRoles
+
+        for (const appId of Object.keys(initialAppsPermissions)) {
+            const initialAppPermissions = initialAppsPermissions[appId].permissions
+            const newAppPermissions = newAppsPermissions[appId].permissions
+
+            for (const permission of Object.keys(initialAppPermissions)) {
+                if (!initialAppPermissions[permission] && newAppPermissions[permission]) {
+                    notificationMessage = UpdatePermissionsMessage
+                }
+            }
+        }
+    }
+
+    return notificationMessage
+}
+
+/**
+ * Updates B2BAppRole objects according table checkboxes.
+ * If user removed "canReadApp" checkbox, then B2BAppRole will be deleted.
+ * If user added "canReadApp" checkbox, then B2BAppRole will be created.
+ */
+const updateB2BAppRolePermissions = async ({
+    employeeRoleId,
+    initialB2bRolePermissions,
+    newB2bRolePermissions,
+    b2BAppPermissions,
+    b2BAppRoles,
+    createB2BAppRoleAction,
+    softDeleteB2BAppRoleAction,
+    updateB2BAppRoleAction,
+}) => {
+    for (const appId of Object.keys(newB2bRolePermissions)) {
+        const initialEmployeePermissionsInB2BApp = initialB2bRolePermissions[appId]
+        const newEmployeePermissionsInB2BApp = newB2bRolePermissions[appId]
+
+        if (!isEqual(initialEmployeePermissionsInB2BApp, newEmployeePermissionsInB2BApp)) {
+            const canReadAppKey = `canRead${appId}`
```

Comment:
What if I have a permission B2BAppPermission with this key? Seems like naming collision